File: higherOrderFun/higherorderfunc.py
# Higher-order function: takes a function as an argument
def apply_function(x, func):
    return func(x)

# Example functions
double_it = lambda n: n * 2
square_it = lambda n: n * n

print(apply_function(5, double_it))  # Output: 10
print(apply_function(4, square_it))  # Output: 16

File: higherOrderFun/higherorderfunc.java
object HigherOrderFunctionExample {
  // Higher-order function: takes a function as a parameter
  def applyFunction(x: Int, func: Int => Int): Int = {
    func(x)
  }

  def main(args: Array[String]): Unit = {
    val doubleIt = (n: Int) => n * 2
    val squareIt = (n: Int) => n * n

    println(applyFunction(5, doubleIt)) // Output: 10
    println(applyFunction(4, squareIt)) // Output: 16
  }
}
